core/src/main/java/de/danoeh/antennapod/core/service/download/DownloadServiceNotification.java

```diff
@@ -107,29 +107,28 @@ public void updateReport(List<DownloadStatus> reportQueue) {
         for (DownloadStatus status : reportQueue) {
             if (status.isSuccessful()) {
                 successfulDownloads++;
+                createReport = createReport || showAutoDownloadReport && !status.isInitiatedByUser() && status.getFeedfileType() == FeedMedia.FEEDFILETYPE_FEEDMEDIA;
             } else if (!status.isCancelled()) {
-                createReport = true;
                 failedDownloads++;
+                createReport = true;
             }
         }
 
         if (createReport) {
             Log.d(TAG, "Creating report");
+
             // create notification object
-            NotificationCompat.Builder builder = new NotificationCompat.Builder(context,
-                    NotificationUtils.CHANNEL_ID_ERROR)
-                    .setTicker(context.getString(R.string.download_report_title))
-                    .setContentTitle(context.getString(R.string.download_report_content_title))
-                    .setContentText(
-                            String.format(
-                                    context.getString(R.string.download_report_content),
-                                    successfulDownloads, failedDownloads)
-                    )
-                    .setSmallIcon(R.drawable.stat_notify_sync_error)
-                    .setContentIntent(
-                            ClientConfig.downloadServiceCallbacks.getReportNotificationContentIntent(context)
-                    )
-                    .setAutoCancel(true);
+            boolean autoDownloadReport = failedDownloads == 0;
+            String channelId = autoDownloadReport ? NotificationUtils.CHANNEL_ID_AUTO_DOWNLOAD : NotificationUtils.CHANNEL_ID_ERROR;
+            int titleId = autoDownloadReport ? R.string.auto_download_report_title : R.string.download_report_title;
+
+            NotificationCompat.Builder builder = new NotificationCompat.Builder(context, channelId);
+            builder.setTicker(context.getString(titleId))
+                   .setContentTitle(context.getString(R.string.download_report_content_title))
```

Comment:
Please use the same string for ticker and content title. Also, I think it would be easier to read the code if you copied the notification builder code: `if(autoDownloadReport){...}else{...}` instead of using the `?` operator in basically every line.

Reply:
Agreed, done.